src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -242,6 +242,21 @@ public function withoutGlobalScopes(?array $scopes = null)
         return $this;
     }
 
+    /**
+     * Remove all except passed registered global scopes.
+     *
+     * @param  array  $scopes
+     * @return $this
+     */
+    public function withoutGlobalScopesExcept(array $scopes = [])
+    {
+        $this->withoutGlobalScopes(
+            array_diff(array_keys($this->scopes), $scopes)
+        );
+
+        return $this;
```

Comment:
You can shorten this with `tap()` if you like:
```suggestion
        return tap($this)->withoutGlobalScopes(
            array_diff(array_keys($this->scopes), $scopes)
        );
```

Reply:
Ironically, you didn't keep it consistent by moving the `array_keys()` inside the `withoutGlobalScopes()` call instead of making it above.